openghg/transform/flux/_edgar.py

```diff
@@ -1,21 +1,162 @@
+"""
+Parse EDGAR flux files and perform area-conservative regridding
+of fluxes.
+
+Currently based on acrg.name.emissions_helperfuncs.getedgarannualtotals()
+
+Additional edgar functions which could be incorporated.
+- getedgarv5annualsectors
+- getedgarv432annualsectors
+- (monthly sectors?)
+
+TODO: Work out how to select frequency
+- could try and relate to period e.g. "monthly" versus "yearly" etc.
+
+EDGAR flux files tend to have dimensions (lat, lon), and no time dimension;
+however, time is included in the filename, so we try to extract this, along
+with version, species, sector, and resolution.
+
+Typical file names:
+
+ v432_CH4_1978.0.1x0.1.nc (or .zip)
+ v50_CH4_1978.0.1x0.1.nc (or .zip)
+ v6.0_CH4_1978_TOTALS.0.1x0.1.nc
+
+ v50_CO2_excl_short-cycle_org_C_1978.0.1x0.1.nc (or .zip)
+ v50_CO2_org_short-cycle_C_1978.0.1x0.1.nc (or .zip)
+ v50_N2O_1978.0.1x0.1.zip (or .zip)
+
+Futher information is contained in the EDGAR readme files.
+For instance, from "_readme.html" from v6.0 data:
+
+'Yearly Emissions gridmaps in ton substance / 0.1degree x 0.1degree / year
+ for the .txt files with longitude and latitude coordinates referring to
+ the low-left corner of each grid-cell.'
+
+'Monthly Emissions gridmaps in ton substance / 0.1degree x 0.1degree / month
+ for the .txt files with longitude and latitude coordinates referring to
+ the low-left corner of each grid-cell.'
+
+'Emissions gridmaps in kg substance /m2 /s for the .nc files with longitude
+ and latitude coordinates referring to the cell center of each grid-cell.'
+
+For EDGAR v8.0,
+
+"""
+
+import pathlib
 import re
 import zipfile
-from pathlib import Path
-from typing import Dict, Optional, Tuple, Union, cast
-from zipfile import ZipFile
+from collections import namedtuple
+from typing import Any, Dict, Optional, Tuple, Union, cast
 import logging
 import numpy as np
 import xarray as xr
 from numpy import ndarray
 
+from openghg.standardise.meta import assign_flux_attributes, define_species_label
+from openghg.store import infer_date_range
+from openghg.util import (
+    clean_string,
+    molar_mass,
+    synonyms,
+    find_coord_name,
+    convert_internal_longitude,
+    timestamp_now,
+)
+
+
 logger = logging.getLogger("openghg.transform.flux")
 logger.setLevel(logging.DEBUG)  # Have to set level for logger as well as handler
 
+
 ArrayType = Optional[Union[ndarray, xr.DataArray]]
+Path = Union[pathlib.Path, zipfile.Path]
+
+
+_edgar_known_versions = ("v432", "v50", "v60", "v70", "v80")
+
+
+# TODO: make this work for...
+# - yearly - "v6.0_CH4_2015_TOTALS.0.1x0.1.nc"
+# - sectoral - "v6.0_CH4_2015_ENE.0.1x0.1.nc"
+# - monthly sectoral - "v6.0_CH4_2015_1_ENE.0.1x0.1.nc", "v6.0_CH4_2015_2_ENE.0.1x0.1.nc", ...
+def assemble_edgar_metadata(
+    filepath: Path,
```

Comment:
This changes goes with removing the type alias
```suggestion
    filepath: Union[pathlib.Path, zipfile.Path],
```